File: 11INFORMATIKA/DATASCIENCE/11KOM2/meet02j/car.py

class Car:

    def __init__(self, make, model, year):
        self.make = make
        self.model = model
        self.year = year
        self.odometer = 0

    def describe(self):
        print(f"This car made by {self.make} model : {self.model}  release in {self.year}")

class Battery:

    def __init__(self):
        self.make = "ABC Company"
        self.capacity = 100_000

    def check_capacity(self):
        print(f"Capacity : {self.capacity} kWh")


class ElectricCar(Car):

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.battery = Battery()


my_toyota = Car("Toyota", "Agya", 2023)
my_toyota.describe() 

my_hyundai = ElectricCar("Hyundai", "KonaEv", 2020)
my_hyundai.describe()
my_hyundai.battery.check_capacity()

    

File: 10INFORMATIKA/10IPA4/Meet08/car.cpp
#include <iostream>
using namespace std;

struct Car {
    string merk;
    string model;
    int year;
    long odometer = 0; // DEFAULT ATTRIBUTE

    Car(string me, string mo, int y) : merk(me), model(mo), year(y){};

    void describe(){
        cout << "This car made by " << merk << " model " << model << " year " << year << endl;
    }
};

struct Battery {
    string name;
    double capacity = 75;
    Battery(string n) : name(n) {};
    void show_capacity(){
        cout << "Battery : " << capacity << "%.\n";
    }
};
struct ElectricCar : Car {
    Battery battery;
    ElectricCar(string me, string mo, int y, string bn) : Car(me, mo, y), battery(bn) {};

};
int main(){
    Car my_toyota("Toyota", "Avanza", 2024);
    my_toyota.describe();

    ElectricCar my_tesla("Tesla", "S", 2024, "Panasonic");
    my_tesla.describe();
    my_tesla.battery.show_capacity();

    return 0;
}